streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -2626,16 +2629,16 @@ object ZStream extends ZStreamPlatformSpecificConstructors with Serializable {
 
   object Pull {
     val end: Pull[Any, Nothing, Nothing]                         = IO.fail(None)
-    def emit[A](a: A): Pull[Any, Nothing, A]                     = UIO.succeed(a)
-    def fail[E](e: E): Pull[Any, E, Nothing]                     = IO.fail(Some(e))
-    def halt[E](c: Cause[E]): Pull[Any, E, Nothing]              = IO.halt(c.map(Some(_)))
-    def die(t: Throwable): Pull[Any, Nothing, Nothing]           = UIO.die(t)
-    def dieMessage(m: String): Pull[Any, Nothing, Nothing]       = UIO.dieMessage(m)
-    def done[E, A](e: Exit[E, A]): Pull[Any, E, A]               = IO.done(e.mapError(Some(_)))
-    def fromPromise[E, A](p: Promise[E, A]): Pull[Any, E, A]     = p.await.mapError(Some(_))
+    def emit[A](a: => A): Pull[Any, Nothing, A]                  = UIO.succeed(a)
+    def fail[E](e: => E): Pull[Any, E, Nothing]                  = IO.fail(Some(e))
+    def halt[E](c: => Cause[E]): Pull[Any, E, Nothing]           = IO.halt(c.map(Some(_)))
+    def die(t: => Throwable): Pull[Any, Nothing, Nothing]        = UIO.die(t)
+    def dieMessage(m: => String): Pull[Any, Nothing, Nothing]    = UIO.dieMessage(m)
+    def done[E, A](e: => Exit[E, A]): Pull[Any, E, A]            = IO.done(e.mapError(Some(_)))
+    def fromPromise[E, A](p: => Promise[E, A]): Pull[Any, E, A]  = p.await.mapError(Some(_))
```

Comment:
Not sure this one should be lazy.

Reply:
I believe this should be. I do require a second opinion.